42/README.md

```diff
@@ -0,0 +1,71 @@
+```
+shortname: [42/HARD-FORKS]
+name: Handling Forks
```

Comment:
The name of this BEP forces a particular solution strategy rather than identifying a problem, surveying possible solutions, and recommending a particular solution. There are two problems, really, which this BEP currently calls Type 1 and Type 2. It's probably okay to include both in one BEP. Maybe the BEP title should be something like "Handling new transaction models and storage schemas".

I'm not sure that the proposed solutions (as I understand them) are correctly described as "hard forks." When I picture a blockchain hard fork, I imagine going back to a previous block and rewriting the chain from there, according to new rules. The old chain and the new chain can be pictured like a [tuning fork](https://en.wikipedia.org/wiki/Tuning_fork).

Reply:
The difference between soft and hard forks is determined by [backwards-compatiblity - it's not about chain rewrites](https://en.wikipedia.org/wiki/Fork_(blockchain)#Hard_fork).

Approach 1 here can be used to handle both soft and hard forks.

`Handling new transaction models and storage schemas` sounds like a good name to me.